### ISSUE TITLE
[glad] Build error on x64-windows

### ISSUE THREAD

**Issue by AutumnGoose** [microsoft/vcpkg#40786]
Copy issue body from E:/vcpkg/installed/vcpkg/issue_body.md

E:\>vcpkg install glad
Computing installation plan...
The following packages will be built and installed:
    glad[core,loader]:x64-windows@0.1.36
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:7890".
Compiler found: E:/Visiual Studio/Visiual Studio IDE/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\1\AppData\Local\vcpkg\archives in 99.2 us. Use --debug to see more details.
Installing 1/1 glad[core,loader]:x64-windows@0.1.36...
Building glad[core,loader]:x64-windows@0.1.36...
-- Note: glad only supports static library linkage. Building static library.
-- Using cached Dav1dde-glad-1ecd45775d96f35170458e6b148eb0708967e402.tar.gz.
-- Cleaning sources at E:/vcpkg/buildtrees/glad/src/708967e402-274d415499.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg/downloads/Dav1dde-glad-1ecd45775d96f35170458e6b148eb0708967e402.tar.gz
-- Applying patch encoding.patch
-- Applying patch find_python.patch
-- Using source at E:/vcpkg/buildtrees/glad/src/708967e402-274d415499.clean
-- This version of glad uses the compatibility profile. To use the core profile instead, create an overlay port of this with GLAD_PROFILE set to 'core' or set GLAD_PROFILE to 'core' in a custom triplet.
-- This recipe is at E:/vcpkg/ports/glad
-- See the overlay ports documentation at https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: E:/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: E:/vcpkg/buildtrees/glad/x64-windows-dbg
    See logs for more information:
      E:\vcpkg\buildtrees\glad\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/glad/portfile.cmake:143 (vcpkg_cmake_install)
  scripts/ports.cmake:192 (include)

error: building glad:x64-windows failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle glad:x64-windows: 3.8 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+glad
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[glad]+Build+error+on+x64-windows&body=Copy+issue+body+from+E%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```[tasklist]
### Tasks
```

**Comment by AutumnGoose** [comment]
Change Dir: 'E:/vcpkg/buildtrees/glad/x64-windows-dbg'

Run Build Command(s): "E:/Visiual Studio/Visiual Studio IDE/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j17 install
[1/4] C:\WINDOWS\system32\cmd.exe /C "cd /D E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean && E:\vcpkg\downloads\tools\python\python-3.11.8-x64\python.exe -m glad --profile=compatibility --out-path=E:/vcpkg/buildtrees/glad/x64-windows-dbg --api= --generator=c-debug --extensions= --spec=gl --reproducible"
FAILED: include/glad/glad.h src/glad.c E:/vcpkg/buildtrees/glad/x64-windows-dbg/include/glad/glad.h E:/vcpkg/buildtrees/glad/x64-windows-dbg/src/glad.c 
C:\WINDOWS\system32\cmd.exe /C "cd /D E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean && E:\vcpkg\downloads\tools\python\python-3.11.8-x64\python.exe -m glad --profile=compatibility --out-path=E:/vcpkg/buildtrees/glad/x64-windows-dbg --api= --generator=c-debug --extensions= --spec=gl --reproducible"
[09/04/2024 03:19:53][INFO	][glad   	]: reproducible build, using packaged specification: 'gl.xml'
[09/04/2024 03:19:53][INFO	][glad.files	]: opening 'gl.xml' from packaged path
Traceback (most recent call last):
  File "xml\etree\ElementTree.py", line 1525, in __init__
  File "xml\parsers\expat.py", line 4, in <module>
ImportError: DLL load failed while importing pyexpat:  Ҳ   ָ    ģ 顣

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "xml\etree\ElementTree.py", line 1528, in __init__
ImportError: DLL load failed while importing pyexpat:  Ҳ   ָ    ģ 顣

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\__main__.py", line 192, in <module>
    main()
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\__main__.py", line 151, in main
    spec = get_spec(spec_name, reproducible=ns.reproducible)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\__main__.py", line 43, in get_spec
    return spec_cls.from_file(glad.files.open_local(value + '.xml', encoding='utf-8-sig'))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\parse.py", line 74, in from_file
    return cls(xml_frompath(path))
               ^^^^^^^^^^^^^^^^^^
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\parse.py", line 20, in xml_frompath
    return etree.parse(path).getroot()
           ^^^^^^^^^^^^^^^^^
  File "xml\etree\ElementTree.py", line 1219, in parse
  File "xml\etree\ElementTree.py", line 575, in parse
  File "xml\etree\ElementTree.py", line 1530, in __init__
ImportError: No module named expat; use SimpleXMLTreeBuilder instead
ninja: build stopped: subcommand failed.

**Comment by AutumnGoose** [comment]
Package: glad[core,loader]:x64-windows@0.1.36

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.33.31630.0
-    vcpkg-tool version: 2024-08-01-fd884a0d390d12783076341bd43d77c3a6a15658
    vcpkg-scripts version: unknown

**To Reproduce**

`vcpkg install glad`

**Failure logs**

```
-- Note: glad only supports static library linkage. Building static library.
-- Using cached Dav1dde-glad-1ecd45775d96f35170458e6b148eb0708967e402.tar.gz.
-- Cleaning sources at E:/vcpkg/buildtrees/glad/src/708967e402-274d415499.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg/downloads/Dav1dde-glad-1ecd45775d96f35170458e6b148eb0708967e402.tar.gz
-- Applying patch encoding.patch
-- Applying patch find_python.patch
-- Using source at E:/vcpkg/buildtrees/glad/src/708967e402-274d415499.clean
-- This version of glad uses the compatibility profile. To use the core profile instead, create an overlay port of this with GLAD_PROFILE set to 'core' or set GLAD_PROFILE to 'core' in a custom triplet.
-- This recipe is at E:/vcpkg/ports/glad
-- See the overlay ports documentation at https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: E:/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: E:/vcpkg/buildtrees/glad/x64-windows-dbg
    See logs for more information:
      E:\vcpkg\buildtrees\glad\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/glad/portfile.cmake:143 (vcpkg_cmake_install)
  scripts/ports.cmake:192 (include)



```

<details><summary>E:\vcpkg\buildtrees\glad\install-x64-windows-dbg-out.log</summary>

```
Change Dir: 'E:/vcpkg/buildtrees/glad/x64-windows-dbg'

Run Build Command(s): "E:/Visiual Studio/Visiual Studio IDE/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j17 install
[1/4] C:\WINDOWS\system32\cmd.exe /C "cd /D E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean && E:\vcpkg\downloads\tools\python\python-3.11.8-x64\python.exe -m glad --profile=compatibility --out-path=E:/vcpkg/buildtrees/glad/x64-windows-dbg --api= --generator=c-debug --extensions= --spec=gl --reproducible"
FAILED: include/glad/glad.h src/glad.c E:/vcpkg/buildtrees/glad/x64-windows-dbg/include/glad/glad.h E:/vcpkg/buildtrees/glad/x64-windows-dbg/src/glad.c 
C:\WINDOWS\system32\cmd.exe /C "cd /D E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean && E:\vcpkg\downloads\tools\python\python-3.11.8-x64\python.exe -m glad --profile=compatibility --out-path=E:/vcpkg/buildtrees/glad/x64-windows-dbg --api= --generator=c-debug --extensions= --spec=gl --reproducible"
[09/04/2024 03:19:53][INFO	][glad   	]: reproducible build, using packaged specification: 'gl.xml'
[09/04/2024 03:19:53][INFO	][glad.files	]: opening 'gl.xml' from packaged path
Traceback (most recent call last):
  File "xml\etree\ElementTree.py", line 1525, in __init__
  File "xml\parsers\expat.py", line 4, in <module>
ImportError: DLL load failed while importing pyexpat: �Ҳ���ָ����ģ�顣

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "xml\etree\ElementTree.py", line 1528, in __init__
ImportError: DLL load failed while importing pyexpat: �Ҳ���ָ����ģ�顣

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\__main__.py", line 192, in <module>
    main()
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\__main__.py", line 151, in main
    spec = get_spec(spec_name, reproducible=ns.reproducible)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\__main__.py", line 43, in get_spec
    return spec_cls.from_file(glad.files.open_local(value + '.xml', encoding='utf-8-sig'))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\parse.py", line 74, in from_file
    return cls(xml_frompath(path))
               ^^^^^^^^^^^^^^^^^^
  File "E:\vcpkg\buildtrees\glad\src\708967e402-274d415499.clean\glad\parse.py", line 20, in xml_frompath
    return etree.parse(path).getroot()
           ^^^^^^^^^^^^^^^^^
  File "xml\etree\ElementTree.py", line 1219, in parse
  File "xml\etree\ElementTree.py", line 575, in parse
  File "xml\etree\ElementTree.py", line 1530, in __init__
ImportError: No module named expat; use SimpleXMLTreeBuilder instead
ninja: build stopped: subcommand failed.
```
</details>

**Comment by sfhacker** [comment]
God save us all!

**Comment by FrankXie05** [comment]
@AutumnGoose Could you please check if there is this file: `pyexpat.pyd` in your `"C:\vcpkg\downloads\tools\python\python-3.11.5-x64\python.exe"`  folder?
It is necessary dll for python on windows.
If you have multiple versions of Python, make sure to use the same version.

**Comment by AutumnGoose** [comment]
@FrankXie05 It do works. There are five different version of python in my Windows.
Due to I still need those different version of python to solve other works, my solution was downloads glad in https://glad.dav1d.de and msys to manage my library.
Deeply thankful for your support. Words cannot fully express my gratitude.

**Comment by YunchengLiu** [comment]
I encountered the same problem. I found the `pyexpat.pyd`  file in my directory `E:\_devenv\vcpkg\downloads\tools\python\python-3.11.8-x64` . But it showed that importing the file still failed. I want to know how to solve this problem? I cloned a vcpkg instance again and the same error occurred
[install-x64-windows-dbg-out.log](https://github.com/user-attachments/files/17075744/install-x64-windows-dbg-out.log)

**Comment by YunchengLiu** [comment]
@FrankXie05 I have 5 python environments hosted by conda. I don't have any directly exposed python versions in my local environment, I manage all my development environments through conda, of course I can't rule out the indirect impact of blender, VS, etc. hosting their own python versions, but at least I don't have any settings for Python specified in my environment variables.

**Comment by dg0yt** [comment]
Does pyexpat.pyd need to find an expat DLL?

**Comment by AutumnGoose** [comment]
@YunchengLiu Your situation is very similar to mine! I use conda to manage my python environment as well, but my VSCode still use another python version. I failed to assign specific any of my python to install glad. I had tried to download vcpkg again but was of no use. So I finally had to download and config glad manually. However, I can only compile and run my code in cmd and a little bit bother.

**Comment by YunchengLiu** [comment]
@AutumnGoose 

I temporarily solved this problem by adding a `python 11._pth` file in the same directory as `python.exe` (managed by vcpkg) to force overwriting the import environment in `sys.path`.

The `python11.__pth` file was created based on the official template file (if you are not familiar with the role of this file, you can refer to [the official introduction of Python](https://docs.python.org/3.12/library/sys_path_init.html#pth-files)), and then the source path of `glad's` build tree was added to this file, which ensured that both `glad` and standard library files can be searched.

After doing this, `vcpkg` successfully built `glad`. After the build was successful, I deleted this file to avoid affecting the build of other libraries.

I noticed that when `vcpkg` downloads embedded python, it will delete the accompanying `._pth` file(https://github.com/microsoft/vcpkg/blob/3d89599850c8168fa4b560367b06c1be52645b20/scripts/cmake/vcpkg_find_acquire_program(PYTHON3).cmake#L27), which may be to fine-tune the behavior of python when it is executed.

Therefore, I guess this `glad` build problem may be related to the path setting of python during build (sorry, I haven't made time to carefully track the build process)

**Comment by psmerk** [comment]
May be some other software (it was Spyder 6 in our case, but it could also be some standard Python distribution) "registered" some particular instance/installation of Python in Windows registry as a "system python". The problem then is that vcpkg's "embedded" python is not so embedded (i.e. isolated) as one might expect, but it consults the registry settings and somehow gets mixed up with the "system python", if it finds any. Under HKEY_LOCAL_MACHINE\SOFTWARE\Python\PythonCore\ there are "system python"s for version.subversion, e.g., 3.11, 3.12 etc. But if Spyder registered its 3.11.9 as "system python" 3.11 and then vcpkg used 3.11.8, such a mix failed. The simplest solution then is to delete the registry settings for 3.11.

(The funny thing was that the use [or whatever] of "system python" could not be traced from the build logs; there was not the slightest mention of the python from the .../spyder/... path in the logs. Nor was Spyder's python anywhere in the PATH, etc. Only when we renamed the spyder folder, vcpkg started to work correctly, but for a long time we had no idea how the vcpkg's python could even know about Spyders' one. :-)

**Comment by YunchengLiu** [comment]
I found that there is indeed a registry item in my registry as @psmerk  said. The `PythonPath` is set to the directory hosted by `miniconda3`, and the python version is inconsistent with what `vcpkg` needs. I deleted this registry item, cleaned the build cache of `glad` and rebuilt `glad`, and the problem was solved.

This is really confusing. I really didn't pay special attention to this behavior of `miniconda3`, or it is difficult to constrain the behavior of each software that hosts python by itself.

I think `vcpkg` should clarify the behavior of the embed python used, or at least output key information such as the searched  library path in the log.
